src/app/services/poke-type.service.ts

```diff
@@ -0,0 +1,28 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { environment } from '../environment/environment';
+import { Efficacy } from '../models/efficacy';
+import { Type } from '../models/type';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PokeTypeService {
+
+  constructor(
+    private http: HttpClient
+  ) { }
+  
+  // Get every type in the table
+  getMany(){
+    return this.http.get<Type[]>(`${environment.apiUrl}/Types`)
+  }
+
+  // Get efficacy of attack given attacker and defender as parameters
+  getEff(att?: string, def?: string) {
```

Comment:
Similar here with the name. This one is debatable and if you left it the same it'd be fine, but ideally you shouldn't need to look things up. Eff might not be immediately apparent and over time it can create jargon in the codebase that adds to a new developer's ramp up time.